### ISSUE TITLE
Error parsing TOML on 1.1.386 if using comments in Arrays

### ISSUE THREAD

**Issue by pavel-naumenko** [microsoft/pyright#9296]
### Description
After upgrading from 1.1.385 to 1.1.386 pyright raises errors "Pyproject file parse attempt 1 error:"

Issue is raised if pyproject.toml has comments in arrays 

E.g. error in 1.1.386 (while OK in 1.1.385
```
[project]
name = "test"
version = "1.0.0"
dependencies = [
    "lib_a==1.1.1", # comment
    "lib_b==1.1.2"
]
```

If removing comments, parsing is OK in new version also
```
[project]
name = "test"
version = "1.0.0"
dependencies = [
    "lib_a==1.1.1", 
    "lib_b==1.1.2"
]
```

**Comment by erictraut** [comment]
Looks like I'll need to back out the change for #9286. 

@InSyncWithFoo, I looked at `smol-toml`, but it unfortunately distributed as an ES Module rather than a CommonJS package. That means there's no straightforward way to use it in pyright.

I'll investigate other options.

**Comment by linde12** [comment]
Can confirm. Even comments above a list is failing for me

**Comment by erictraut** [comment]
I switched to a different toml parser library — one that claims to support the toml 1.0 standard. Let's see if this addresses all of the issues.

**Comment by Dr-Irv** [comment]
@erictraut I'm having the same issue with `pandas-stubs`, so if you use that as a test, you can see if the new TOML library handles it.  Our `pyproject.toml` file has lots of comments in it.

**Comment by erictraut** [comment]
@Dr-Irv, good idea. I confirmed that the new toml parser works fine with `pandas-stubs`.

**Comment by plu-bgl** [comment]
I saw this one too late, but I created #9339 and I think the same issue was introduced in v1.1.285: https://github.com/microsoft/pyright/issues/4367. It's all about using `CRLF` vs `LF` or `CR`.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.387.
